src/librustdoc/lib.rs

```diff
@@ -117,21 +117,9 @@ fn get_args() -> Option<Vec<String>> {
         .collect()
 }
 
-fn stable<F>(name: &'static str, f: F) -> RustcOptGroup
-where
-    F: Fn(&mut getopts::Options) -> &mut getopts::Options + 'static,
-{
-    RustcOptGroup::stable(name, f)
-}
-
-fn unstable<F>(name: &'static str, f: F) -> RustcOptGroup
-where
-    F: Fn(&mut getopts::Options) -> &mut getopts::Options + 'static,
-{
-    RustcOptGroup::unstable(name, f)
-}
-
 fn opts() -> Vec<RustcOptGroup> {
+    let stable: fn(_, fn(&mut getopts::Options) -> &mut _) -> _ = RustcOptGroup::stable;
+    let unstable: fn(_, fn(&mut getopts::Options) -> &mut _) -> _ = RustcOptGroup::unstable;
```

Comment:
Maybe it'd more clear/less verbose to directly import constructors via `use`?
```suggestion
    use RustcOptGroup::{stable, unstable};
    
```

Reply:
I think it's fine as is but I'll let it up to @jyn514.

Reply:
That doesn't work because it's an associated function unfortunately:

```
error[E0432]: unresolved import `RustcOptGroup`
   --> src/librustdoc/lib.rs:125:9
    |
125 |     use RustcOptGroup::{stable, unstable};
    |         ^^^^^^^^^^^^^ `RustcOptGroup` is a struct, not a module
```

Reply:
Ah, that makes sense. Thanks and sorry for the noise!